src/test/java/redis/clients/jedis/tests/ConnectionTest.java

```diff
@@ -48,4 +50,32 @@ public void checkCloseable() {
     client.connect();
     client.close();
   }
-}
\ No newline at end of file
+
+  @Test
+  public void getErrorAfterConnectionReset() throws Exception {
+    brokenReason(new byte[1024 * 1024 + 1][0], "ERR Protocol error: invalid multibulk length");
+    brokenReason(new byte[1][512 * 1024 * 1024 + 1], "ERR Protocol error: invalid bulk length");
```

Comment:
I don't think we should test two scenarios here. We'e not testing redis behaviour, we just want to test that Jedis handles connection reset with remaining body correct. With just one test case it's more than enough to validate that code.